saleor/webhook/payloads.py

```diff
@@ -1080,3 +835,530 @@ def generate_excluded_shipping_methods_for_checkout_payload(
+):
+    serializer = PayloadSerializer()
+
+    user_fields = ("email", "first_name", "last_name")
+    channel_fields = ("slug", "currency_code")
+    shipping_method_fields = ("name", "type", "currency", "price_amount")
+
+    # todo use the most appropriate warehouse
+    warehouse = None
+    if checkout.shipping_address:
+        warehouse = Warehouse.objects.for_country(
+            checkout.shipping_address.country.code
+        ).first()
+
+    extra_dict_data = {
+        # Casting to list to make it json-serializable
+        "included_taxes_in_price": include_taxes_in_prices(),
+        "lines": serialize_checkout_lines(checkout, lines, get_line_prices_data),
+        "collection_point": json.loads(
+            _generate_collection_point_payload(checkout.collection_point)
+        )[0]
+        if checkout.collection_point
+        else None,
+        "meta": generate_meta(requestor_data=generate_requestor(requestor)),
+    }
+
+    extra_dict_data.update(checkout_prices_data)
+
+    checkout_data = serializer.serialize(
+        [checkout],
+        fields=CHECKOUT_FIELDS,
+        obj_id_name="token",
+        additional_fields={
+            "channel": (lambda o: o.channel, channel_fields),
+            "user": (lambda c: c.user, user_fields),
+            "billing_address": (lambda c: c.billing_address, ADDRESS_FIELDS),
+            "shipping_address": (lambda c: c.shipping_address, ADDRESS_FIELDS),
+            "shipping_method": (lambda c: c.shipping_method, shipping_method_fields),
+            "warehouse_address": (
+                lambda c: warehouse.address if warehouse else None,
+                ADDRESS_FIELDS,
+            ),
+        },
+        extra_dict_data=extra_dict_data,
+    )
+    return checkout_data
+
+
+def _get_line_prices_data_with_taxes(
+    checkout_info: CheckoutInfo,
+    manager: PluginsManager,
+    lines: Iterable[CheckoutLineInfo],
+    line_info: CheckoutLineInfo,
+) -> Dict[str, Decimal]:
+    currency = checkout_info.checkout.currency
+    unit_price_data = checkout_calculations.checkout_line_unit_price(
+        manager=manager,
+        checkout_info=checkout_info,
+        lines=lines,
+        checkout_line_info=line_info,
+        discounts=[],
+    )
+    unit_price = quantize_price(unit_price_data.price_with_sale, currency)
+    unit_price_with_discounts = quantize_price(
+        unit_price_data.price_with_discounts, currency
+    )
+
+    return {
+        "price_net_amount": unit_price.net.amount,
+        "price_gross_amount": unit_price.gross.amount,
+        "price_with_discounts_net_amount": unit_price_with_discounts.net.amount,
+        "price_with_discounts_gross_amount": unit_price_with_discounts.gross.amount,
+    }
+
+
+def _get_line_prices_data_without_taxes(
+    checkout: "Checkout",
+    line_info: CheckoutLineInfo,
+) -> Dict[str, Decimal]:
+    return {
+        "price_net_amount": quantize_price(
```

Comment:
Why net amount instead of "base"?

Reply:
We already have a "base" price field in checkout lines, it's the amount specified in channel listing.

Reply:
Isn't that exactly the one that we need (plus a discount)?